File: backend/app/domain/entity/about.py
from pydantic import BaseModel
from typing import Optional
from datetime import date


class About(BaseModel):
    id: int
    name: str
    title: str
    summary: str
    profile_image: str
    bio: str


class Education(BaseModel):
    id: int
    about_id: Optional[int] = None
    institution: str
    degree: str
    field: str
    start_date: date
    end_date: Optional[date] = None
    description: Optional[str] = None


class Experience(BaseModel):
    id: int
    about_id: Optional[int] = None
    company: str
    position: str
    start_date: date
    end_date: Optional[date] = None
    description: Optional[str] = None
    achievements: Optional[list[str]] = None


class SocialMedia(BaseModel):
    id: int
    about_id: Optional[int] = None
    platform: str
    url: str
    username: Optional[str] = None


class AboutResponse(BaseModel):
    about: About
    education: list[Education]
    experience: list[Experience]
    social_media: list[SocialMedia]

File: frontend/src/lib/types/about.ts
export interface About {
  id: number;
  name: string;
  title: string;
  summary: string;
  profile_image: string;
  bio: string;
}

export interface Education {
  id: number;
  about_id?: number | null;
  institution: string;
  degree: string;
  field: string;
  start_date: string;
  end_date?: string | null;
  description?: string | null;
}

export interface Experience {
  id: number;
  about_id?: number | null;
  company: string;
  position: string;
  start_date: string;
  end_date?: string | null;
  description?: string | null;
  achievements?: string[] | null;
}

export interface SocialMedia {
  id: number;
  about_id?: number | null;
  platform: string;
  url: string;
  username?: string | null;
}

export interface AboutResponse {
  about: About;
  education: Education[];
  experience: Experience[];
  social_media: SocialMedia[];
} 
